### PR TITLE
Upgrade debian-base to 0.3.1 for CVEs

### PR DESCRIPTION
**Pull request by satyasm**
**What this PR does / why we need it**:
Upgrade debian-base to 0.3.1 in response to CVE fixes in debian-base

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Bumps up the version number to v11.

**Release note**:

```release-note
Version increment to v11 caused by upgrade to debian-base 0.3.1 in response to CVE fixes.
```

### CODE REVIEW

**Comment by satyasm** [comment]
output from local `make build`

```
Successfully built 30bce0e0aca6
Successfully tagged staging-k8s.gcr.io/debian-iptables-amd64:v11
```

so 0.3.1 has been pushed to production.

**Comment by cblecker** [comment]
/hold
There are more places where this needs to be updated: https://cs.k8s.io/?q=debian-base-&i=fosho&files=&repos=kubernetes

**Comment by satyasm** [comment]
This is for the networking components. I believe there will be other PRs for the other components.

**Comment by cblecker** [comment]
It also appears that the debian-base Makefile itself hasn't been updated: https://github.com/kubernetes/kubernetes/blob/b90485560185085ae7a8ad3fab60cb417f7654bf/build/debian-base/Makefile#L21

**Comment by tallclair** [comment]
@cblecker This is really highlighting the friction in keeping addon images up-to-date. Right now there is too much of a disconnect between the opensource manifests, and the google-internal process for actually publishing those to k8s.gcr.io. I'm following up with the google releng team to see how we can fix this, but in the meantime the priority is to get the new images rolled out to fix some CVEs in debian-base. I asked @satyasm to just update all debian-base references to 0.3.1 for now.

**Comment by satyasm** [comment]
@tallclair, @cblecker updated all the files. thanks for catching this.

File: build/debian-hyperkube-base/Makefile [hunk 1]
```diff
@@ -23,7 +23,7 @@ TAG=0.10
 ARCH?=amd64
 CACHEBUST?=1
 
-BASEIMAGE=k8s.gcr.io/debian-base-$(ARCH):0.3
+BASEIMAGE=k8s.gcr.io/debian-base-$(ARCH):0.3.1
```
Comment (ixdy) [inline build/debian-hyperkube-base/Makefile]: should this also update the tag for debian-hyperkube-base?

Comment (satyasm) [reply build/debian-hyperkube-base/Makefile]: good catch, bumped to 0.11

File: test/images/resource-consumer/BASEIMAGE [hunk 2]
```diff
@@ -1,4 +1,4 @@
-amd64=k8s.gcr.io/debian-base-amd64:0.3
-arm=k8s.gcr.io/debian-base-arm:0.3
-arm64=k8s.gcr.io/debian-base-arm64:0.3
-ppc64le=k8s.gcr.io/debian-base-ppc64le:0.3
+amd64=k8s.gcr.io/debian-base-amd64:0.3.1
+arm=k8s.gcr.io/debian-base-arm:0.3.1
+arm64=k8s.gcr.io/debian-base-arm64:0.3.
```
Comment (ixdy) [inline test/images/resource-consumer/BASEIMAGE]: missing `1` here

Comment (satyasm) [reply test/images/resource-consumer/BASEIMAGE]: ouch, don't know how I missed that. good catch and thanks! updated.

**Comment by satyasm** [comment]
@ixdy thanks for the reviews, updated as noted.

**Comment by satyasm** [comment]
testing seems to be stuck. Going to close this and re-open as a new PR. @ixdy will assign the new one to you once I have it open.

**Comment by satyasm** [comment]
This has been reopened as #67026

**Pull request closed without merging**